src/deterministic.jl

```diff
@@ -18,29 +18,100 @@ function Deterministic(
     data::Union{Dict{Dates.DateTime, Any}, SortedDict{Dates.DateTime, Any}};
     normalization_factor::NormalizationFactor = 1.0,
     scaling_factor_multiplier::Union{Nothing, Function} = nothing,
-    timestamp = :timestamp,
     resolution::Union{Dates.Period, Nothing} = nothing,
 )
     for (k, v) in data
-        if v isa DataFrames.DataFrame
-            data[k] = TimeSeries.TimeArray(v; timestamp = timestamp)
-        elseif v isa TimeSeries.TimeArray
+        if isa(v, DataFrames.DataFrame)
+            data[k] = Array(v)
+        elseif isa(v, TimeSeries.TimeArray)
+            data[k] = TimeSeries.values(v)
+            if resolution === nothing
+                resolution = TimeSeries.timestamp(v)[2] - TimeSeries.timestamp(v)[1]
+            end
+        elseif isa(v, Vector{Float64})
             continue
         else
             try
-                data[k] =
-                    TimeSeries.TimeArray(range(k, length = length(v), step = resolution))
+                data[k] = Vector{Float64}(v...)
             catch e
-                throw(ArgumentError("The values in the data dict can't be converted to TimeArrays. Resulting error: $e"))
+                throw(ArgumentError("The values in the data dict are in an invalid format. Resulting error: $e"))
```

Comment:
You're probably better of rethrowing the exception.  What you have could be an error log message.